examples/ct/ct_handlers_test.go

```diff
@@ -357,6 +358,48 @@ func TestAddChain(t *testing.T) {
 	assert.Equal(t, "BAEABnNpZ25lZA==", resp.Signature)
 }
 
+// Submit a chain with a valid precert but not complete path to root. Should be rejected.
+func TestAddPrecertChainMissingIntermediate(t *testing.T) {
+
+}
+
+// Submit a chain with a valid path but using a cert instead of a precert. Should be rejected.
+func TestAddPrecertChainCert(t *testing.T) {
+	km := new(crypto.MockKeyManager)
+	client := new(trillian.MockTrillianLogClient)
+
+	roots := loadCertsIntoPoolOrDie(t, []string{testonly.CACertPEM})
+	reqHandlers := CTRequestHandlers{0x42, roots, client, km, time.Millisecond * 500, fakeTimeSource}
+
+	cert, err := fixchain.CertificateFromPEM(testonly.TestCertPEM)
+	_, ok := err.(x509.NonFatalErrors)
```

Comment:
I see. So, t.Fatal below occurs if (a) there was an err and (b) it wasn't a x509.NonFatalErrors.

So, since you parsed TestCertPEM (which is not a pre-cert) you definitely don't expect an x509.NonFatalErrors err to be returned.

However if TestCertPEM turned out to be a pre-cert the test would continue.

I think what I'm saying is that you'd want t.Fatal called if it turned out that TestCertPEM _was_ a precert.

In other words, the !ok part is superfluous.